Feature/2020support

thomas-gleiforst opened the pull request.

BenFRC5147 committed:
Expands Pwr/Bms

Adds 12 new Board Enable/Disable buttons. Also adds feedback for whether or not a button is actually enabled or disabled

amwgvf committed:
added heading

I added heading to MapViewModel and Map Model

amwgvf committed:
changed map compass

I changed the rover point on the map from a dot to an arrow

ColtonJacobson23 committed:
In Progress - First Commit

Needed to commit to have code up on GitHub
- The selector is in the view
- Currently modifying the DataGridTextColumns to start hiding them on selection changes
- Working out bindings for each DGTC so that the DMS has each binding per column

ColtonJacobson23 committed:
List Viewing Complete

- Selecting an option in the DMS selector changes the columns underneath to the correct options

BenFRC5147 committed:
Fixed enable/disable notifier

ConfusedPolarBear committed:
First embedded digital camera test

ConfusedPolarBear committed:
Refactor camera code

ConfusedPolarBear committed:
Alpha watchdog functionality

ConfusedPolarBear committed:
Add second camera

ConfusedPolarBear committed:
Bad multiplexer

ConfusedPolarBear committed:
Rewrote camera multiplexer

Changed MjpegDecoder implementations to a newer fork with a more efficient frame decoder.

ConfusedPolarBear committed:
Working camera multiplexer

ConfusedPolarBear committed:
Fixed watchdog breaking the camera streams

Also integrated the screenshot code inside the multiplexer

ConfusedPolarBear committed:
Code cleanup

thomas-gleiforst committed:
Updated BaseStation IP

Changed base address from 131.151.19.185 to 192.168.1.50

ColtonJacobson23 committed:
DMS Still In PRogress

thomas-gleiforst committed:
Possibly improved switching and screenshotting

The index that was being screenshotted and unsubscirbed was never changed properly. This ties the senders index Tag to the index and runs off that.

thomas-gleiforst committed:
Rotate cameras on center press

Quick and easy implementation of what may eventually be changing settings through rovecomm

ConfusedPolarBear committed:
Fix surface removal and improved rotation

ConfusedPolarBear committed:
Add 4th camera

ConfusedPolarBear committed:
Support multiple camera boards in the multiplexer

ConfusedPolarBear committed:
Add camera enum and deprecate old functions

ConfusedPolarBear committed:
Add more cameras to RAM

chellewarz committed:
telemetry feature

telemetry feature

thomas-gleiforst committed:
Merge branch 'feature/ExpandedPWR&BMS' into feature/2020support

thomas-gleiforst committed:
TEMP invert longitude because N3

This will be reverted, but the N3 sends the negative longitude since that is what it did in the past. To ensure the N3 isn't messed up between now and SAR, basestation will temporarily invert it. This WILL be undone.

thomas-gleiforst committed:
Update for 2020 Rovecomm IDs

I think everything is updated properly according to the manifest in github. Notes: Nav has nothing. Drive and Lighting are seperate but the same board ids. Some of the descriptions suck either on my end or on the document end, but I believe these to be all correct. No errors are implemented.

thomas-gleiforst committed:
Fixed Rover model not working

thomas-gleiforst committed:
First round of rovecomm updates

BMS, Power and Lighting updated

thomas-gleiforst committed:
duplicated left/right drive

thomas-gleiforst committed:
Power and BMS changes

Currently 2020 rover has 2020 power board and 2019 bms. So commented out this years bms, readded last years bms, and changed their ips to account for a conflict we observed.

thomas-gleiforst committed:
Rovecomm updates round 2

This should be all of the rovecomm updates that were easy, and the MetadataManagerConfig has been updated with comments as to what has been implemented and what has not.

thomas-gleiforst committed:
Updated gimbal

I skipped gimbal because it needed input fixes, which have been carried out. There is a chance the left joystick controls the right gimbal, but I believe this to be correct.

thomas-gleiforst committed:
Added n3 data

Only added critical n3 data since the rest seems to be not confirmed

thomas-gleiforst committed:
Significant Science Sensor updates

thomas-gleiforst committed:
Drive commands corrected, readded bms telem

Forgot to change the speed left right commands that we discovered during testing the embedded cameras. Also removed a little too much telemetry from BMS, so readded temperature and pack current

ConfusedPolarBear committed:
Add first pass at proper camera MVVM

BenFRC5147 committed:
Adds a button for Vacuuming

Adds the button 'A' to enable Vacuum, vacuum is enabled while the button is held and disabled otherwise.

BenFRC5147 committed:
Adds buttons for chemicals

Binds buttons X,Y, and B to Chemicals 1,2, and 3 respectively

thomas-gleiforst committed:
Updated current display

Most things from the power board are connected. Some were just accidentally commented out, and I just implemented the science based ones.

thomas-gleiforst committed:
Merge branch 'feature/2020support' of https://github.com/MissouriMRDT/BaseStation_Software into feature/2020support

BenFRC5147 committed:
Update InputManagerConfig.cs

Additional commit to actually bind XYB

BenFRC5147 committed:
Merge branch 'feature/2020support' of https://github.com/MissouriMRDT/BaseStation_Software into feature/2020support

BenFRC5147 committed:
Screw -> Zactuation

Makes the changes to have Screw be Zactuation

BenFRC5147 committed:
Deletes the Actuation MVVM

As title, because as Thomas says "I think we don't need those anymore"

amwgvf committed:
fixed graphs

I added new graphs for methane, co2, and o2 and I added labels on the axes

amwgvf committed:
Merge branch 'feature/2020support' of https://github.com/MissouriMRDT/BaseStation_Software into feature/2020support

thomas-gleiforst committed:
Removed the remnants of Science Actuation

ConfusedPolarBear committed:
Add more XAML work

rkbhkp committed:
Geneva Rovecomm implementation

Added rovecomm functionality to geneva to spin the motor left and right by clicking the buttons on geneva

amwgvf committed:
clean up spectrometer view

added MPPC graph and made a drop down to toggle between spectrometer and mppc

amwgvf committed:
Merge branch 'feature/2020support' of https://github.com/MissouriMRDT/BaseStation_Software into feature/2020support

rkbhkp committed:
Science Sensors  fix and clean up

Changed over the way the packes are gathered to the new way using a list for science sensors and changed the sensors to display the new values in the box.

amwgvf committed:
cleaning up spectrometer view

I cleaned mppc plot model and added axes labels

amwgvf committed:
Merge branch 'feature/2020support' of https://github.com/MissouriMRDT/BaseStation_Software into feature/2020support

thomas-gleiforst committed:
Commented out spectrometer awaits

thomas-gleiforst committed:
Updated packet grabbing and float casting

Sometimes we were making Int16 arrays when we were being sent floats, we are still dividing by 1000 like when we sent int16s, and some places still used the old rovecomm packet grabbing structure.

thomas-gleiforst committed:
Signed byte update

An unchecked byte is a byte where the overflow is ignored, wheres an sbyte is a signed byte

thomas-gleiforst committed:
Updates 30V power telemetry

thomas-gleiforst committed:
Controller and Rovecomm updates

Updated controller schemes, fixed the button mapping issue in science (calling it valuepulse rather than vacuum pulse), and logged which rovecomm changes have been tested.

rkbhkp committed:
Fixed Science Sensor cutting off words

Fixed the columns cutting off the characters in science sensors

thomas-gleiforst committed:
Merge branch 'dev' into feature/2020support

thomas-gleiforst committed:
Removed pitch/heading/roll from gps

For some reason this didn't go through in the merge.

thomas-gleiforst committed:
Made code slightly more future proof

Arithmetic is a better bet in the long run than a manual if statement

thomas-gleiforst committed:
Removed old camera views

thomas-gleiforst committed:
Merge branch 'feature/2020support' into feature/digital-cameras

thomas-gleiforst committed:
Moved camera files to core

Having the MVVM in RED will make it harder for RAM to embed cameras

thomas-gleiforst committed:
Added cameras everywhere

thomas-gleiforst committed:
Removed a bit more legacy code

thomas-gleiforst committed:
Moving camera files back around because Caliburn

skeltonn committed:
Fixed issue with 2020/2019 flip

thomas-gleiforst committed:
Merge pull request #95 from MissouriMRDT/feature/digital-cameras

Feature/digital cameras

skeltonn committed:
Motor bus enables and 2019 BMS support

skeltonn committed:
Merge branch 'feature/2020support' of https://github.com/MissouriMRDT/BaseStation_Software into feature/2020support

BenFRC5147 committed:
Fixed images and timer size

BenFRC5147 committed:
Merge branch 'feature/2020support' of https://github.com/MissouriMRDT/BaseStation_Software into feature/2020support

thomas-gleiforst committed:
Swap gimbal and nav ids

thomas-gleiforst committed:
Merge branch 'feature/2020support' of https://github.com/MissouriMRDT/BaseStation_Software into feature/2020support

thomas-gleiforst committed:
Fixed 3D model rotation

We had the rover set to rotate to the pitch, yaw, and roll of the rover. Moving to the yaw made it impossible to figure out what was being depicted (as noted, just implemented incorrectly). I also realized it makes more sense to just send rotation packets when we get the packets rather than when the variables are updated. This calls the function less. Finally, the orientation of the N3 on the rover means that -pitch and -roll is what the rover should be rotated based off of.

thomas-gleiforst committed:
Removed crashing when getting long/lat

thomas-gleiforst committed:
Fixed text strikeout when not fixed

Currently this year doesn't show fix, so crossing out lat/long when not fixed is annoying.

BenFRC5147 committed:
Stops console expansion

Stops the console from expanding when you have too many messages in it.

rkbhkp committed:
Changed Martial to partial pressure

Fixed the "martial pressure" to the correct partial pressure and fixed what it says the sensors view values are measured in.

BenFRC5147 committed:
Updates Camera View

Puts the Camera feed below the controls as opposed to above them.

amwgvf committed:
added mppc stuff

I duplicated a lot of the spectrometer functions and changed them to mppc so now we have both

amwgvf committed:
Merge branch 'feature/rotating_compass' into feature/2020support

amwgvf committed:
fixed the rotating arrow

the arrow on the map now rotates roughly around its center

rkbhkp committed:
adjusted red formating

fixed camera formating problems and problems with red layout

thomas-gleiforst committed:
Slowed Down Gimbals

thomas-gleiforst committed:
Removed compass in RED

BenFRC5147 committed:
Adjusts Margins

Adjusts margins and Widths to make RED look like the best RED it can be.

BenFRC5147 committed:
Merge branch 'feature/2020support' of https://github.com/MissouriMRDT/BaseStation_Software into feature/2020support

amwgvf committed:
Changed the updating of spectrometer and mppc

I changed the way we update the spectrometer and mppc graphs so they were more like the sensor graphs

amwgvf committed:
Moved Sensor Graph Buttons

I moved the start sensor graph and update sensor graph buttons from the sensor section to the sensor graph section

amwgvf committed:
adjusted spectrometer and sensor graphs

connected the buttons to their functions and added a download button

thomas-gleiforst committed:
Updated annoatations

In sensor graphs, the annotations always went to the main graph, rather than appliying the the graph that is actually selected.

thomas-gleiforst committed:
Merge branch 'feature/2020support' of https://github.com/MissouriMRDT/BaseStation_Software into feature/2020support

ConfusedPolarBear committed:
Remove integer parameters from CameraMultiplexer

Added error handling to the screenshot process

thomas-gleiforst committed:
Moved around spectrometer buttons

BenFRC5147 committed:
Renames RNM

Renames all of the RNM stuff to be not RNM but instead RON

EliVerbrugge reviewed (approved):
The camera board IPs in CameraMultiplexer.cs should probably be referenced from the Manifest.

BenFRC5147 committed:
Adds the Rover Switch to Ping tool

Adds the rover switch to the ping tool (as well as network map) Requires a deletion of REDConfigs.

Also Fixes the Rover's 2.4Mhz rocket, which, while probably very good for NLOS, would probably be much more appreciated as the proper 2.4Ghz

Renames RO:N to RO-N

antrob25 committed:
Nano Brain

antrob25 committed:
Nano Brain

EliVerbrugge committed:
Constructing ArmConsole now

EliVerbrugge committed:
Constructing ArmConsole now

EliVerbrugge committed:
Merge branch 'feature/Arm_Upgrades' of https://github.com/MissouriMRDT/BaseStation_Software into feature/Arm_Upgrades

EliVerbrugge committed:
Fixed config issue

thomas-gleiforst committed:
Metadata description and comment updates

thomas-gleiforst committed:
Update restart from RED

thomas-gleiforst committed:
TEMP Update to consistenly drive gimbals

In order to make the gimbal control logical from the controller, we have to negate the values in a confusing way. This should be redone to send all positives and have the electrical system handle it.

thomas-gleiforst committed:
Updated Science Actuation and Sensors

1) The vacuum only has one direct on/off packet from power board
2) Spectrometer never inherited rovecomm receiver
3) Science graph was still parsing in some of the data the old way
4) Science sensors and graphs weren't even watching for the packets they needed
5) I believe the pressures from science sensors are in mBar

thomas-gleiforst committed:
Converting chemical control to int16s

According to the board software, it claims that the chemical motors are looking for an int16. I assume it wants a number between 0 and 1000, like just the positive half of a drive motor. The rovecomm manifest originally said it should just be looking for a 0 or 1 byte, and while this has been updated, exactly what it is looking for hasn't changed.

thomas-gleiforst committed:
Fixed compile issues

thomas-gleiforst committed:
Completed autonomy telemetry - not connected with Rovecomm

thomas-gleiforst committed:
Merge branch 'feature/autonomy_updates' into feature/2020support

thomas-gleiforst committed:
Updated autonomy telemetry with rovecomm changes

thomas-gleiforst committed:
Fixed csproj merge issue

thomas-gleiforst committed:
Formatting updates

thomas-gleiforst committed:
Merge branch 'feature/Arm_Upgrades' into feature/2020support

thomas-gleiforst committed:
Fixed csproj merge issue

amwgvf committed:
fixing site management

it works now

amwgvf committed:
Merge branch 'feature/2020support' of https://github.com/MissouriMRDT/BaseStation_Software into feature/2020support

thomas-gleiforst committed:
Finished DMS view and implementation

thomas-gleiforst committed:
Remove unneeded functions

thomas-gleiforst committed:
Merge branch 'feature/DMS' into feature/2020support

thomas-gleiforst committed:
Remove outdated call and DMS view selector

thomas-gleiforst committed:
Merge branch 'feature/digital-cameras' into feature/2020support

thomas-gleiforst committed:
Create science files directory if i
